txbridge/src/main/java/org/jboss/jbossts/txbridge/outbound/OutboundBridgeManager.java

```diff
@@ -66,18 +68,20 @@
 
         try
         {
-            Transaction transaction = (Transaction)TransactionManager.transactionManager().getTransaction();
+            Transaction transaction = (Transaction)TransactionManager.transactionManager(new InitialContext()).getTransaction();
 
-            Uid externalTxId = transaction.get_uid();
+            // generating new Uid to get it connected to incoming transaction
+            // the incoming transaction could come from different TM than Narayana is (e.g. Wildfly Transaction client)
+            Uid outboundMappingUid = new Uid();
```

Comment:
@tomjenkinson after tries to bound uid from client as you suggests I'm again rethinking the approach. When checking all dependencies in project I think it's not a good idea to depend on wildfly transaction client in Narayana directly. XTS should be 

What about the approach like this?
```
// maps JTA Tx Id to OutboundBridge instance.
private static final ConcurrentMap<Transaction, Uid> transctionToUidMappings = new ConcurrentHashMap<>();
private static final ConcurrentMap<Uid, org.jboss.jbossts.txbridge.outbound.OutboundBridge> outboundBridgeMappings = new ConcurrentHashMap<>();

...
   
transctionToUidMappings.putIfAbsent(transaction, new Uid());
Uid outboundMappingUid = transctionToUidMappings.get(transaction);

if(!outboundBridgeMappings.containsKey(outboundMappingUid)) {
    createMapping(transaction, outboundMappingUid);
}
```